readthedocs/projects/forms.py

```diff
@@ -634,6 +634,19 @@ def clean_domain(self):
             domain_string = parsed.netloc
         else:
             domain_string = parsed.path
+
+        if (
+            settings.PRODUCTION_DOMAIN and
+            settings.PRODUCTION_DOMAIN in domain_string
+        ):
+            raise forms.ValidationError('Enter a valid domain name.')
```

Comment:
We should mention exactly why this is invalid (`{PRODUCTION_DOMAIN} is not a valid domain`